Replicator/ReplicatorOptions.hh

```diff
@@ -36,20 +39,24 @@ namespace litecore { namespace repl {
         PropertyDecryptor       propertyDecryptor       {nullptr};
         void*                   callbackContext         {nullptr};
         std::atomic<C4ReplicatorProgressLevel> progressLevel {kC4ReplProgressOverall};
-        bool                    collectionAware         {true};
-
+        mutable bool            collectionAware         {true};
```

Comment:
I think this should be a method, rather than a forcibly mutable item in an otherwise mutable struct.  The logic can be the same logic used to calculate this method.

Reply:
Then it would need a big note, or it should be moved out to the Replicator, as its value will not be valid or computed before calling verify() function (I also commented about this in another place).